### PR TITLE
Fix 10115 - Wokflow Calculate Action broken under PHP8

### PR DESCRIPTION
**Pull request by markbond1007**
Added a check for when request id is set but set to blank.

## Description

Currently when trying to set up a workflow with a calculate field action under PHP the form doesnt not generate, 

1) Create a new workflow
2) Choose a module
3) Add an Action
4) Choose Calculate Field
5) The form to fill in does not appear when using PHP8

## Motivation and Context

Fixes broken code under PHP8

## How To Test This

1) Create a new workflow
2) Choose a module
3) Add an Action
4) Choose Calculate Field
5) The form to fill will appear after the fix is applied

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by chris001** [comment]
@markbond1007 Shouldn't the branch be set to [`hotfix`](https://github.com/salesagility/SuiteCRM/tree/hotfix), not [`hotfix-7.13.x`](https://github.com/salesagility/SuiteCRM/tree/hotfix-7.13.x)?  And all 64 of your commits squashed to one commit?

**Comment by pgorod** [comment]
Those 64 commits aren't his... he just needs to bring his branch up-to-date with the commits in the main repo, so that they don't all show inside this PR

**Comment by markbond1007** [comment]
@pgorod 

I directly forked this from master, made the update and pull requested it back to the hotfix 7.13.x as per the instructions, when looking at my fork it tells me i am 'ahead' of SuiteCRM:hotfix, I dont know why its comparing to hotfix as I forked master and pulled req'd to hotfix-7.13.x

**Comment by chris001** [comment]
@markbond1007 
Try and change this PR's branch to merge into, from `hotfix-7.13.x` to `hotfix` only.  See if that clears this up.

**Comment by pgorod** [comment]
I think what should be done (and I believe the instructions say this) is to start your branch from the branch you will be directing your PR towards. And make sure it is up to date _before_ you start adding your own commits.

So typically this is what I would do:
- I have my commits somewhere where I am working on this
- I finish my work
- I go and bring my SuiteCRM up to date with SalesAgility's, specifically the branch (like `hotfix-7.13`) where I aim to add my stuff
- after it is up to date, I start a new branch from there (not from master)
- I move my commits on top of that (just by editing the files, or using git stash, or using cherry-pick from another branch
- I create the pull request

**Comment by markbond1007** [comment]
LOL, well the docs I find today are definately different to those I followed a couple of days back (cant even find them now).

I dont do enough forking to follow that exactly, so I just followed the instructions on Github (salesagilities/Suite instructions), I cant find them now but I notice the ones I find now are pointed at  the SuiteCRM docs site,

@pgorod thats basically what I did, I forked a brand new repo, made changes and the pull request'd it back (with steps inbetween about creating branches based on bug numbers).

The issue was the instructions I found said to fork from master and then go back to hotfix 7.13.x, the instructions say to send it back to hotfix (on SuiteCRM website).

**Comment by markbond1007** [comment]
Well what I found was a version of this:

https://github.com/salesagility/SuiteCRM/blob/master/.github/CONTRIBUTING.md

**Comment by markbond1007** [comment]
@pgorod 
Yes well I knew what I meant with forking branches ;-) they key is the document I found was waaaay out of date...

File: modules/AOW_WorkFlow/controller.php [hunk 1]
```diff
@@ -549,7 +549,7 @@ protected function action_getAction()
 
         $id = '';
         $params = array();
-        if (isset($_REQUEST['id'])) {
+        if (isset($_REQUEST['id']) && strlen(trim($_REQUEST['id'])) > 0 ) {
```
Comment (clemente-raposo) [inline modules/AOW_WorkFlow/controller.php:552]: Hi @markbond1007, would `!empty($_REQUEST['id']))` do the same as `isset($_REQUEST['id']) && strlen(trim($_REQUEST['id'])) > 0`?

Comment (markbond1007) [reply modules/AOW_WorkFlow/controller.php:552]: @clemente-raposo 

Yes that would work just as well (just tried it and its fine).

Mark

**Review by clemente-raposo — APPROVED** [review]

**Review by johnM2401 — APPROVED** [review]

**Pull request merged**